### ISSUE TITLE
how to just rebuild files in current project? --nodeps doesn't seem to do that 

### ISSUE THREAD

**Issue by timotheecour** [dlang/dub#944]
dub build --nodeps --verbose:
still takes about 5 seconds (when no files need to be recompiled); and shows a bunch of actions like:

vibe-d:web 0.7.29: target for configuration "library" is up to date.

Generate target vibe-d:mongodb (staticLibrary /home/timothee/.dub/packages/vibe-d-0.7.29/vibe-d vibe-d_mongodb)

which seem to actually do what I would expect nodeps not to do

expected outcome: only run dmd + linking on files in current project, don't do anything on dependencies, and take < 0.1 seconds when no rebuild is needed

**Comment by s-ludwig** [comment]
`--nodeps` just skips the implicit "upgrade" step that is done to collect/download missing dependencies. However, that gets already skipped as long as all dependencies are satisfied, so it doesn't save time in the common case. Ignoring changes in dependencies is not possible, but it may make sense to implement a directory watcher daemon at some point that makes checking for changes a quick O(1) operation.

But 5 seconds sounds like a lot. Do you use an SSD or HDD, or is there some dependency with a big number of source files? I get about 500 ms for a typical no-op build (Linux). An optimized DUB build unfortunately segfaults for me, but I'd guess that that would bring that number down considerably.

**Comment by timotheecour** [comment]
dependencies:
```
 "ggplotd": ">=0.4.5",
 "msgpack-rpc": "~>0.1.2",
  "dproto": "~>2.1.0",
 "cpuid": "~>0.2.6"
```

log:
```
time dub build --nodeps
Performing "debug" build using dmd for x86_64.
cpuid 0.2.6: target for configuration "library" is up to date.
painlesstraits 0.0.3: target for configuration "library" is up to date.
dproto 2.1.0: target for configuration "library" is up to date.
derelict-util 2.1.0: target for configuration "library" is up to date.
derelict-ft 1.0.2: target for configuration "library" is up to date.
x11 1.0.15: target for configuration "library" is up to date.
cairod 0.0.1-alpha.3+1.10.2: target for configuration "static-custom-posix" is up to date.
dstats 1.0.3: target for configuration "library" is up to date.
dunit 1.0.12: target for configuration "library" is up to date.
gtk-d:gtkd 3.3.1: target for configuration "library" is up to date.
ggplotd 1.1.0: target for configuration "ggplotd-gtk" is up to date.
msgpack-d 0.9.6: target for configuration "library" is up to date.
vibe-d:utils 0.7.29: target for configuration "library" is up to date.
vibe-d:data 0.7.29: target for configuration "library" is up to date.
vibe-d:core 0.7.29: target for configuration "libevent" is up to date.
vibe-d:http 0.7.29: target for configuration "library" is up to date.
vibe-d:diet 0.7.29: target for configuration "library" is up to date.
vibe-d:mail 0.7.29: target for configuration "library" is up to date.
vibe-d:mongodb 0.7.29: target for configuration "library" is up to date.
vibe-d:redis 0.7.29: target for configuration "library" is up to date.
vibe-d:web 0.7.29: target for configuration "library" is up to date.
vibe-d 0.7.29: target for configuration "libevent" is up to date.
msgpack-rpc 0.1.2: target for configuration "default" is up to date.
lhd ~master: target for configuration "application" is up to date.
To force a rebuild of up-to-date targets, run again with --force.
```
dub build --nodeps  4.52s user 0.08s system 99% cpu 4.605 total

What about another option `--ignoredepschanges` (or whatever name, doesn't matter), to really ignore this step?
This is a common use case during development, the user knows what he's doing, he's only developping his files and wants speedy rebuilds

**Comment by s-ludwig** [comment]
The question is if it's the change checking that takes up the time. Can you post another log with `--vverbose`? That will print the time stamps, too.

**Comment by timotheecour** [comment]
```
dub build --nodeps --vverbose &> /tmp/z01.txt
```

shows no timestamps in log

what else should i try?

**Comment by s-ludwig** [comment]
Oh sorry, that's my bad. I actually just had some local modifications that output a time stamp for diagnostic messages, but I wasn't content about them yet. Can you add `output.writef("%s ", Clock.currTime(UTC()));` before the following line and re-run? https://github.com/dlang/dub/blob/1824d16056a680fcca52d3ce87fb43d3679ff3ee/source/dub/internal/vibecompat/core/log.d#L79

**Comment by dhasenan** [comment]
tldr: can close.

I checked this out on a project with 23 dependencies. Built the same target twice. The first time took about 45 seconds for rebuilding dependencies. On the second time:

```
2017-Sep-16 15:28:24.5081531Z Performing "debug" build using dmd for x86_64.
2017-Sep-16 15:28:24.5084858Z d2sqlite3 0.13.1: target for configuration "with-lib" is up to date.
...
2017-Sep-16 15:28:24.5295506Z Copying target from /home/dhasenan/prog/vibe-thrift/.dub/build/library-debug-linux.posix-x86_64-dmd_2076-C2152D0D0A7EBF31E354BBF98BA655BE/libvibe-thrift.a to /home/dhasenan/prog/vibe-
thrift
2017-Sep-16 15:28:24.5301904Z Target '/home/dhasenan/prog/old/stratos/.dub/build/application-debug-linux.posix-x86_64-dmd_2076-D3D880F505117523E453EDFAAFD21193/stratos' doesn't exist, need rebuild.
```

About 30 milliseconds total, a bit more than one millisecond per dependency. I don't think we need to worry about this.

**Comment by MartinNowak** [comment]
Indeed, I've optimized a couple of things that were slow.
I'm also not worrying too much about stat calls in giant code-bases atm.
